### PR TITLE
Implementation of Machine Modes

### LINKED ISSUE #356: Hermetic Casing II; no recipe?
### GregTech CEu Version

1.0.11b-166

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

Unsure

### Expected Behavior

I expected "gtceu:mv_hermetic_casing"/Hermetic Casing II to have recipe. 

### Actual Behavior

I go to check the recipe for the casing, and nothing shows up. I do not know recipe so I also do not know if it can currently be crafted, I would assume not though..? 

### Steps to Reproduce

Check recipe of the Hermetic Casing II with JEI, nothing shows.

### Additional Information

thx

### LINKED ISSUE #358: NullPointerException when accessing MachineDefinition.asStack() in GTCreativeModeTabs
### GregTech CEu Version

1.0.1.1.b-build_171

### Recipe Viewer Installed

REI

### Environment

Singleplayer

### Cross-Mod Interaction

Unsure

### Expected Behavior

Not Crash.

### Actual Behavior

Crash when open the gregtech machine creative tab.

### Steps to Reproduce

1. Open the gregtech machine creative tabs.

### Additional Information

[Crash log](https://crashy.net/e069065d-1e6f-49f6-b3b9-738a36a0add5)

### PR DESCRIPTION
**Pull request by Rundas01**
Huge shoutout to screret for fixing the stuff I was unable to fix

### CODE REVIEW

**Review by screret — APPROVED** [review]
lgtm

File: common/src/main/java/com/gregtechceu/gtceu/api/registry/registrate/MachineBuilder.java [hunk 1]
```diff
@@ -91,8 +93,9 @@ public class MachineBuilder<DEFINITION extends MachineDefinition> extends Builde
     private Consumer<ItemBuilder<? extends MetaMachineItem, ?>> itemBuilder;
     @Setter
     private NonNullConsumer<BlockEntityType<BlockEntity>> onBlockEntityRegister = MetaMachineBlockEntity::onBlockEntityRegister;
-    @Setter
-    private GTRecipeType recipeType;
+    private GTRecipeType[] recipeTypes;
+    @Persisted
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/api/registry/registrate/MachineBuilder.java:97]: persited is unnecessary

Comment (screret) [reply common/src/main/java/com/gregtechceu/gtceu/api/registry/registrate/MachineBuilder.java:97]: no it isn't that's the active type, which does need to be saved

Comment (Yefancy) [reply common/src/main/java/com/gregtechceu/gtceu/api/registry/registrate/MachineBuilder.java:97]: ?? be saved in the `MachineBuilder`??

Comment (screret) [reply common/src/main/java/com/gregtechceu/gtceu/api/registry/registrate/MachineBuilder.java:97]: oh damn i didn't look.

**Pull request merged**